internal/server/providers/azure_test.go

```diff
@@ -0,0 +1,328 @@
+					RefreshToken: models.EncryptedAtRest("bbb"),
+					ExpiresAt:    time.Now().Add(time.Minute * -5),
+					LastUpdate:   time.Now(),
+				}
+
+				err = data.UpdateProviderUser(db, pu)
+				assert.NilError(t, err)
+
+				return user
+			},
+			infoResponse: `{
+				"sub": "o_aaabbbccc",
+				"name": "Steve Harrington",
+				"family_name": "Harrington",
+				"given_name": "Steve",
+				"picture": "https://graph.microsoft.com/v1.0/me/photo/$value"
+			}`,
+			verifyFunc: func(t *testing.T, err error, user *models.Identity) {
+				assert.NilError(t, err)
+
+				pu, err := data.GetProviderUser(db, provider.ID, user.ID)
+				assert.Assert(t, err)
+
+				assert.Assert(t, pu.AccessToken != "aaa")
+			},
+		},
+		{
+			name: "deleted user's userinfo response causes sync to fail",
+			setupFunc: func(t *testing.T) *models.Identity {
+				user := &models.Identity{
+					Name: "rbuckleyn@example.com",
+				}
+
+				err = data.CreateIdentity(db, user)
+				assert.NilError(t, err)
+
+				pu := &models.ProviderUser{
+					ProviderID: provider.ID,
+					IdentityID: user.ID,
+
+					Email:        user.Name,
+					RedirectURL:  "http://example.com",
+					AccessToken:  models.EncryptedAtRest("aaa"),
+					RefreshToken: models.EncryptedAtRest("bbb"),
+					ExpiresAt:    time.Now().Add(time.Minute * -5),
+					LastUpdate:   time.Now(),
+				}
+
+				err = data.UpdateProviderUser(db, pu)
+				assert.NilError(t, err)
+
+				return user
+			},
+			infoResponse: `{
+				"sub": "o_aaabbbccc",
+				"picture": "https://graph.microsoft.com/v1.0/me/photo/$value"
+			}`,
+			verifyFunc: func(t *testing.T, err error, user *models.Identity) {
+				assert.ErrorContains(t, err, "could not get user info from provider")
+			},
+		},
+		{
+			name: "failure to sync groups does fail sync",
+			setupFunc: func(t *testing.T) *models.Identity {
+				graphGroupMemberEndpoint = "https://" + serverURL + "/v1.0/me/memberOf/fail"
+
+				user := &models.Identity{
+					Name: "nwheeler@example.com",
+				}
+
+				err = data.CreateIdentity(db, user)
+				assert.NilError(t, err)
+
+				pu := &models.ProviderUser{
+					ProviderID: provider.ID,
+					IdentityID: user.ID,
+
+					Email:        user.Name,
+					RedirectURL:  "http://example.com",
+					AccessToken:  models.EncryptedAtRest("aaa"), // this is used to fail the groups call, in reality this token should be valid
+					RefreshToken: models.EncryptedAtRest("bbb"),
+					ExpiresAt:    time.Now().Add(time.Minute * -5),
+					LastUpdate:   time.Now(),
+				}
+
+				err = data.UpdateProviderUser(db, pu)
+				assert.NilError(t, err)
+
+				return user
+			},
+			infoResponse: `{
+				"sub": "o_aaabbbccc",
+				"sub": "o_aaabbbccc",
+				"name": "Nancy Wheeler",
+				"family_name": "Wheeler",
+				"given_name": "Nancy",
+				"picture": "https://graph.microsoft.com/v1.0/me/photo/$value"
+			}`,
+			verifyFunc: func(t *testing.T, err error, user *models.Identity) {
+				assert.NilError(t, err)
```

Comment:
Anything else we can check here? We expect the `ProviderUser` record to not have changed?